### PR TITLE
Fix name of the github action and add badge for that action

### PR DESCRIPTION
**Pull request by Ricardo-Osorio**

### CODE REVIEW

File: .github/workflows/basic-checks.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-name: Vet and Test
+name: GO get and vet
 
-on: [push, pull_request]
+on: [push]
```
Comment (Mojachieee) [inline .github/workflows/basic-checks.yml:3]: Is there a reason we're not running this on pull request anymore?

Comment (Ricardo-Osorio) [reply .github/workflows/basic-checks.yml:3]: When I opened a PR the same action is run twice and there's no need for that
If we run it on push it runs whether it's a commit into a branch or into an open PR. Making a PR out of a branch doesn't mean we have to rerun action again, there's no point - the code won't change with a PR.

Comment (Mojachieee) [reply .github/workflows/basic-checks.yml:3]: Ah I thought it was required to be ran on a pull request for the required statuses to work, but that might not be the case

**Review by Mojachieee — APPROVED** [review]

**Comment by Ricardo-Osorio** [comment]
Good point but since it seems I can merge this one then that doesn't apply

**Pull request merged**